librepo/downloader.c

```diff
@@ -1123,6 +1123,19 @@ prepare_next_transfer(LrDownload *dd, gboolean *candidatefound, GError **err)
     // Add the transfer to the list of running transfers
     dd->running_transfers = g_slist_append(dd->running_transfers, target);
 
+    // Call LrStartTransferCb to inform upper levels about start of new transfer
+    LrStartTransferCb st_cb = target->target->starttranscb;
+    if (st_cb) {
```

Comment:
Try another approach. If you set callback function then libcurl will call your code just before the connect call.
Hint: `curl_easy_setopt(CURL *handle, CURLOPT_SOCKOPTFUNCTION, sockopt_callback)`

Reply:
done